olive/evaluator/olive_evaluator.py

```diff
@@ -955,10 +955,10 @@ def _inference(
                 outputs = post_func(result)
             else:
                 raise ValueError("Post processing function is required for SNPE model")
-            preds.extend(outputs.tolist())
-            targets.extend(labels.tolist())
-            lg = result["results"].get("logits")
-            logits.extend(lg.to_list() if lg else [])
+            preds.extend(outputs.flatten().tolist())
```

Comment:
should the flatten in post_processing?

Reply:
That could work too but if the shape of output from the snpe inference is not as expected (batch size X sequence length X vocab), then I think it is worth looking at it first. Since we don't have any issue the pytorch and onnx evaluator with the same data.

Reply:
Looks like the output shape from the inference might be (batchsize X 1 X sequence length X vocab) since the original batch size becomes a part of the output shape. 

In that case, for this example we could flatten the first and second dimensions together in the post processing to make it simple. But in the long term, we could perhaps introduce a flag like `first_dim_is_batch` or something similar so that we concatenate along the first dim here https://github.com/microsoft/Olive/blob/main/olive/platform_sdk/qualcomm/snpe/tools/inference.py#L271 instead of stacking them.

Reply:
The accuracy metric only accept 1d-list as input. To achieve that, we can do that in `post_processing` or have a `flatten` by the bottom of evaluation. I chose the latter one in previous implementation.

As for why this is raised only in SNPE case, I think the root course is:
There are two kinds of `batch_size`:
1. from original data loader configs, let us call "batch_size"
![image](https://github.com/microsoft/Olive/assets/13343117/68ad5ebb-7903-48b3-ada1-0d31815c7b5c)
2. from `FileListDataLoader(...).batch_size`, let us call "snpe_batch_size".

In SNPE case, let take `batch_size=2` as an example, 
1. the data loader firstly created by huggingface data loader`(2, sequence_length)`
2. then FileListCommonDataLoader save the numpy array  to local binary file where each file `(2, sequence_length)` contains the one iteration from data loader.
3. `snpe_batch_size` should always be 1(one file per iteration) for this kind of case, but currently, it is (None)[https://github.com/microsoft/Olive/blob/main/olive/evaluator/olive_evaluator.py#L1001C1-L1001C69]. The `None` or any numbers great than 1 for `snpe_batch_size` will expand the input shape to `[snpe_batch_size, batch_size, sequence_length]` that is why [stack](https://github.com/microsoft/Olive/blob/main/olive/platform_sdk/qualcomm/snpe/tools/inference.py#L271) here I think. It organizes the data shape into `[total_input_length/snpe_batch_size, *output_shape]`. For bert, `batch_size, classes_number` is the real output shape described in io_config.




For the flag of `first_dim_is_batch`, I think it might not work very well as that is the output from model where we cannot force user always know that.

Reply:
Maybe a post process function like:

```python
def snpe_post_process(output_data, **kwargs):
    import torch

    logits = torch.tensor(output_data["results"]["logits"])
    logits = logits.reshape(-1, *logits.shape[2:])
    _, preds = torch.max(logits, dim=-1)

    return preds
```

Reply:
Then how about the labels. :(

Reply:
I think it is needed. There are many case user will prepare the input list, they still need this feature. 
`[Jambayk]I added the option for a file list dataloader to have a batch size option so that we can call net run on multiple input at once`

But besides that, I am thinking, maybe we can avoid to expose the `snpe_batch_size` to users. Please help take a look this commit.
https://github.com/microsoft/Olive/pull/925/commits/551b2f359c0817fb66872522d08f4c1518b8a0f6
The idea is: user is only awareness of the single input file, olive can use `snpe_batch_size` internally run in parallel.
That means, for snpe/qnn model, if the input/output shape are specified by users, then in `post_process`, user only know the specified output shape.

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/884614a7-dfd2-4901-bf31-57b69f19a316)